test/automerge-unit.jl

```diff
@@ -45,32 +45,49 @@ end
-        @test !AutoMerge.meets_sequential_version_number(v"0.1.1", v"0.2.1")[1]
-        @test !AutoMerge.meets_sequential_version_number(v"0.1.2", v"0.2.2")[1]
-        @test !AutoMerge.meets_sequential_version_number(v"1.0.1", v"1.1.1")[1]
-        @test !AutoMerge.meets_sequential_version_number(v"1.0.3", v"1.2.0")[1]
-        @test !AutoMerge.meets_sequential_version_number(v"1.0.3", v"1.2.1")[1]
-        @test !AutoMerge.meets_sequential_version_number(v"1.0.3", v"1.1.1")[1]
-        @test !AutoMerge.meets_sequential_version_number(v"1.0.0", v"2.0.1")[1]
-        @test !AutoMerge.meets_sequential_version_number(v"1.0.0", v"2.1.0")[1]
-        @test !AutoMerge.meets_sequential_version_number(v"1.0.0", v"2.1.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"0.0.1"], v"0.0.2")[1]
+        @test AutoMerge.meets_sequential_version_number([v"0.1.0"], v"0.1.1")[1]
+        @test AutoMerge.meets_sequential_version_number([v"0.1.0"], v"0.2.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"1.0.0"], v"1.0.1")[1]
+        @test AutoMerge.meets_sequential_version_number([v"1.0.0"], v"1.1.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"1.0.0"], v"2.0.0")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"0.0.1"], v"0.0.3")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"0.1.0"], v"0.3.0")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"1.0.2")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"1.2.0")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"3.0.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"0.1.1"], v"0.2.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"0.1.2"], v"0.2.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"0.1.3"], v"0.2.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"1.0.1"], v"1.1.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"1.0.2"], v"1.1.0")[1]
+        @test AutoMerge.meets_sequential_version_number([v"1.0.3"], v"1.1.0")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"0.1.1"], v"0.2.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"0.1.2"], v"0.2.2")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.1"], v"1.1.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.3"], v"1.2.0")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.3"], v"1.2.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.3"], v"1.1.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"2.0.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"2.1.0")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"2.1.0")[1]
+
+        @test AutoMerge.meets_sequential_version_number([v"0.0.1", v"0.1.0"], v"0.0.2")[1]
```

Comment:
I believe this test in particular shows the new functionality. The following tests are adapted from https://github.com/JuliaLang/julia/commit/5741c7f53c5ea443bbd73cc5157b83de415c6d1b.